### PR TITLE
Added prior predictive sampling

### PR DESCRIPTION
**Pull request by digicosmos86**
After using a hack mentioned [here](https://discourse.pymc.io/t/pm-sample-prior-predictive-fails-with-incorrect-size-info-passed-to-the-rng-fn-function-of-the-randomvariable/12513/5), the problem is successfully resolved.

We can now use `model.sample_prior_predictive()` to perform prior predictive sampling

### CODE REVIEW

File: src/hssm/distribution_utils/dist.py [hunk 1]
```diff
@@ -117,13 +117,16 @@ class SSMRandomVariable(RandomVariable):
...
         # to get around the support checking in PyMC that would result in error
-        ndim_supp: int = 0
+        ndim_supp: int = 1
 
         ndims_params: list[int] = [0 for _ in list_params]
         dtype: str = "floatX"
         _print_name: tuple[str, str] = ("SSM", "\\operatorname{SSM}")
         _list_params = list_params
 
+        def _supp_shape_from_params(*args, **kwargs):
```
Comment (AlexanderFengler) [inline src/hssm/distribution_utils/dist.py:127]: can you add a comment above this to explain (for future reference) what this does, and link to issue?

File: src/hssm/hssm.py [hunk 2]
```diff
@@ -516,6 +517,37 @@ def sample_posterior_predictive(
...
+        ----------
+        draws
+            Number of draws to sample from the prior predictive distribution. Defaults
+            to 500.
+        var_names
+            A list of names of variables for which to compute the prior predictive
+            distribution. Defaults to ``None`` which means both observed and unobserved
+            RVs.
+        omit_offsets
+            Whether to omit offset terms in the plot. Defaults to ``True``.
```
Comment (AlexanderFengler) [inline src/hssm/hssm.py:539]: plot?

File: tests/test_hssm.py [hunk 3]
```diff
@@ -217,3 +217,18 @@ def test_model_with_approx_differentiable_likelihood_type(data_angle):
     assert model.loglik == download_hf(loglik)
     trace = model.sample()
     assert isinstance(trace, az.InferenceData)
+
+
+def test_sample_prior_predictive(data):
```
Comment (AlexanderFengler) [inline tests/test_hssm.py:222]: Prob wise to add a test that involves multiple regressions, and one that involves hierarchical terms. 
just to make sure nothing funny happens. 

Ok if you disagree, but if so, please add a comment in the code that explains why these tests cover abstractly speaking all cases of interest.

File: src/hssm/distribution_utils/dist.py [hunk 4]
```diff
@@ -160,6 +163,11 @@ def rng_fn(
                 size = args[-1]
                 args = args[:-1]
 
+            # Although we got around the ndims_supp issue, the size parameter passed
+            # here is still an array with one element. We need to take it out.
+            if not np.isscalar(size):
+                size = np.squeeze(size)
```
Comment (AlexanderFengler) [inline src/hssm/distribution_utils/dist.py:169]: Reading this line I went down a bit and also looked at line 185 ++.
Do we report somewhere the complete logic on how we deal with sizes? 
Just for future reference, I think we should record this as comments.

**Review by AlexanderFengler — CHANGES_REQUESTED** [review]
Looks good, left a few comments.

File: src/hssm/distribution_utils/dist.py (hunk 4 above)
Comment (digicosmos86) [reply src/hssm/distribution_utils/dist.py:169]: Added this as Note in the docstring

File: src/hssm/hssm.py (hunk 2 above)
Comment (digicosmos86) [reply src/hssm/hssm.py:539]: This is directly copied from bambi documentation lol. I don't know what this refers to

File: tests/test_hssm.py (hunk 3 above)
Comment (digicosmos86) [reply tests/test_hssm.py:222]: Added tests

File: src/hssm/hssm.py (hunk 2 above)
Comment (AlexanderFengler) [reply src/hssm/hssm.py:539]: prob just take 'plot' out? 
Can add back in if it turns out to make sense :)

**Review by AlexanderFengler — APPROVED** [review]
Left single comment but ready for merging.

File: src/hssm/hssm.py (hunk 2 above)
Comment (digicosmos86) [reply src/hssm/hssm.py:539]: Will do

**Pull request merged**